host/traceh.c

```diff
@@ -314,6 +314,23 @@ oe_result_t oe_log(oe_log_level_t level, const char* fmt, ...)
     return result;
 }
 
+void* _oe_log_context = NULL;
+oe_log_callback _oe_log_callback = NULL;
+
+oe_result_t oe_log_set_callback(void* context, oe_log_callback callback)
+{
+    if (oe_mutex_lock(&_log_lock) == OE_OK)
+    {
+        _oe_log_context = context;
+        _oe_log_callback = callback;
+        oe_mutex_unlock(&_log_lock);
+
+        return OE_OK;
+    }
+
+    return OE_FAILURE;
```

Comment:
This part of the code should be unreachable except for dev error (i.e. calling `oe_mutex_lock()` with an invalid lock pointer) In general, I would encourage everyone to avoid returning errors that are not meaningful in exchange for something that expresses intent or helpful for debugging.
```suggestion
    return OE_UNEXPECTED;
```

Another option here is to capture and return the actual result from the call to `oe_mutex_lock()`

Reply:
8b89a97 seems to have been dropped in your rebase, can you put it back in?

Reply:
Thanks for your notice, now it is back.